src/app/people/people.component.ts

```diff
@@ -50,16 +51,24 @@ export class PeopleComponent implements OnInit {
   @Input() isEditingEnabled: boolean;
   @Output() onChanged = new EventEmitter<any>();
   @Output() onDelete = new EventEmitter<Person>();
-  displayedColumns: string[] = ["personDesc", "availability", "allocated", "unallocated", "assignmentCount", "commitFraction"];
   @ViewChild(MatSort) sort: MatSort;
-  
+
   constructor(public dialog: MatDialog) { }
 
   ngOnInit() {
   }
 
+  displayedColumns(): string[] {
+    for (var i = 0; i < this.people.length; i++) {
```

Comment:
A few things here.

* Firstly, let's please avoid the repetition of elements between the "with location" and "without location" arrays.
* Secondly, instead of this for loop, it's more concise and readable to use the `find()` method on the array, to see if an element exists that matches your requirements.
* Thirdly, let's simply filter on whether the person has a "truthy" location, rather than comparing to the empty string. For example `undefined !== ''` but we don't want `location = undefined` to trigger the location column to appear.

I haven't tested this, but what I had in mind was something like the following:

```
let result = ['personDesc'];
if (this.people.find(p => p.location)) {
    result.push('location');
}
return result.concat(['availability', 'allocated', ...]);
```